### PR TITLE
use tempfile instead of namefixed one

### LINKED ISSUE #243: parallel execution of tailor (e.g. to build / deploy) errors out with `error: Cannot stat ".combined.env"`
https://github.com/opendevstack/tailor/blob/328fcbca12408fd11d359d241bb1f69f3811fd00/pkg/openshift/template.go#L41-L58

bets on a file named `.combined.env` - which is later deleted .. in parallel cases, like the below ...

```
parallel (
  'couper': {
    buildImage(context, "couper", "couper", tailorSelBase+"-couper")
  },
  'acservice': {
    buildImage(context, "acservice", "ac-service", tailorSelBase+"-ac-service")
  },
  'dataservice': {
    buildImage(context, "dataservice", "data-service", tailorSelBase+"-data-service")
  },
  'pptservice': {
    buildImage(context, "pptexportservice", "pptexport-service", tailorSelBase+"-pptexport-service")
  },
  failFast: true
)
```

you'll get an error: 
```
+ tailor --non-interactive -n ahdm-dev apply --selector service=..... --param ODS_OPENSHIFT_APP_DOMAIN=...... --preserve bc:/spec/output/imageLabels --preserve bc:/spec/output/to/name --ignore-unknown-parameters bc,is
error: Cannot stat ".combined.env": stat .combined.env: no such file or directory

Comparing templates in . with OCP namespace ahdm-dev.
```

### PR DESCRIPTION
**Pull request by clemensutschig**
closes #243 for master

### CODE REVIEW

File: pkg/openshift/template.go [hunk 1]
```diff
@@ -47,13 +47,22 @@ func ProcessTemplate(templateDir string, name string, paramDir string, compareOp
...
 			return []byte{}, err
 		}
-		tempParamFile := ".combined.env"
-		defer os.Remove(tempParamFile)
-		cli.DebugMsg("Writing contents of param files into", tempParamFile)
-		err = ioutil.WriteFile(tempParamFile, paramFileBytes, 0644)
+		
+		tempParamFile, err := ioutil.TempFile("", ".combined.*.env")
 		if err != nil {
+			log.Fatal(err)
```
Comment (michaelsauter) [inline pkg/openshift/template.go]: `log` is not imported - better to return the error anyway I think: `return []byte{}, err`

File: pkg/openshift/template.go [hunk 2]
```diff
@@ -47,13 +47,22 @@ func ProcessTemplate(templateDir string, name string, paramDir string, compareOp
...
+		}
+
+		defer os.Remove(tempParamFile.Name())
+
+		cli.DebugMsg("Writing contents of param files into", tempParamFile.Name())
+		err = tempParamFile.Write(paramFileBytes)
```
Comment (michaelsauter) [inline pkg/openshift/template.go:56]: `Write` returns two values

File: pkg/openshift/template.go [hunk 3]
```diff
@@ -47,14 +47,20 @@ func ProcessTemplate(templateDir string, name string, paramDir string, compareOp
 		if err != nil {
 			return []byte{}, err
 		}
-		tempParamFile := ".combined.env"
-		defer os.Remove(tempParamFile)
-		cli.DebugMsg("Writing contents of param files into", tempParamFile)
-		err = ioutil.WriteFile(tempParamFile, paramFileBytes, 0644)
+		
+		tempParamFile,_ := ioutil.TempFile("", ".combined.*.env")
```
Comment (michaelsauter) [inline pkg/openshift/template.go:51]:
```suggestion
		tempParamFile, err := ioutil.TempFile("", ".combined.*.env")
		if err != nil {
				return []byte{}, err
		}
```

**Review by michaelsauter — CHANGES_REQUESTED** [review]
Thanks a lot!

One minor suggested change to return an error - could happen if the temp dir is not writable by the current user.

**Review by michaelsauter — APPROVED** [review]

**Comment by clemensutschig** [comment]
@michaelsauter - do you burn a new release?

**Comment by michaelsauter** [comment]
@clemensutschig Already done: https://github.com/opendevstack/tailor/releases/tag/v1.3.3 :)

I will open PRs for inclusion into the Jenkins base agent as well.

**Pull request merged**